demo/agw/ScrolledThumbnail.py

```diff
@@ -0,0 +1,640 @@
+        self.zoombutton.SetFont(boldFont)
+        self.fontbutton.SetFont(boldFont)
+        self.dirbutton.SetFont(boldFont)
+        self.colourbutton.SetFont(boldFont)
+
+
+    def DoLayout(self):
+
+        splitsizer = wx.BoxSizer(wx.VERTICAL)
+        optionsizer = wx.StaticBoxSizer(self.optionsizer_staticbox, wx.VERTICAL)
+        zoomsizer = wx.BoxSizer(wx.HORIZONTAL)
+        thumbsizesizer = wx.BoxSizer(wx.HORIZONTAL)
+        customsizer = wx.StaticBoxSizer(self.customsizer_staticbox, wx.VERTICAL)
+        thumbsizer = wx.StaticBoxSizer(self.thumbsizer_staticbox, wx.VERTICAL)
+        radiosizer = wx.BoxSizer(wx.VERTICAL)
+        dirsizer = wx.StaticBoxSizer(self.dirsizer_staticbox, wx.HORIZONTAL)
+        dirsizer.Add(self.dirbutton, 0, wx.LEFT|wx.BOTTOM|wx.ALIGN_CENTER_VERTICAL|wx.ADJUST_MINSIZE, 3)
+        splitsizer.Add(dirsizer, 0, wx.EXPAND|wx.TOP|wx.LEFT, 5)
+        radiosizer.Add(self.radiostyle1, 0, wx.LEFT|wx.TOP|wx.ADJUST_MINSIZE, 3)
+        radiosizer.Add(self.radiostyle2, 0, wx.LEFT|wx.TOP|wx.ADJUST_MINSIZE, 3)
+        radiosizer.Add(self.radiostyle3, 0, wx.LEFT|wx.TOP|wx.ADJUST_MINSIZE, 3)
+        radiosizer.Add(self.radiostyle4, 0, wx.LEFT|wx.TOP|wx.BOTTOM|wx.ADJUST_MINSIZE, 3)
+        thumbsizer.Add(radiosizer, 1, wx.EXPAND, 0)
+        splitsizer.Add(thumbsizer, 0, wx.TOP|wx.EXPAND|wx.LEFT, 5)
+        customsizer.Add(self.highlight, 0, wx.LEFT|wx.TOP|wx.BOTTOM|wx.ADJUST_MINSIZE, 3)
+        customsizer.Add(self.showfiles, 0, wx.LEFT|wx.BOTTOM|wx.ADJUST_MINSIZE, 3)
+        customsizer.Add(self.enabledragging, 0, wx.LEFT|wx.BOTTOM|wx.ADJUST_MINSIZE, 3)
+        customsizer.Add(self.setpopup, 0, wx.LEFT|wx.BOTTOM|wx.ADJUST_MINSIZE, 3)
+        customsizer.Add(self.setgpopup, 0, wx.LEFT|wx.BOTTOM|wx.ADJUST_MINSIZE, 3)
+        customsizer.Add(self.enabletooltip, 0, wx.LEFT|wx.BOTTOM|wx.ADJUST_MINSIZE, 3)
+        splitsizer.Add(customsizer, 0, wx.TOP|wx.EXPAND|wx.LEFT, 5)
+        zoomsizer.Add(self.textzoom, 1, wx.ALL|wx.ALIGN_CENTER_HORIZONTAL|wx.ALIGN_CENTER_VERTICAL|wx.ADJUST_MINSIZE, 3)
+        zoomsizer.Add(self.zoombutton, 0, wx.ALL|wx.ALIGN_CENTER_HORIZONTAL|wx.ALIGN_CENTER_VERTICAL|wx.ADJUST_MINSIZE, 3)
+        thumbsizesizer.Add(self.textthumbwidth, 1, wx.ALL|wx.ALIGN_CENTER_HORIZONTAL|wx.ALIGN_CENTER_VERTICAL|wx.ADJUST_MINSIZE, 3)
+        thumbsizesizer.Add(self.textthumbheight, 1, wx.ALL|wx.ALIGN_CENTER_HORIZONTAL|wx.ALIGN_CENTER_VERTICAL|wx.ADJUST_MINSIZE, 3)
+        thumbsizesizer.Add(self.thumbsizebutton, 0, wx.ALL|wx.ALIGN_CENTER_HORIZONTAL|wx.ALIGN_CENTER_VERTICAL|wx.ADJUST_MINSIZE, 3)
+        optionsizer.Add(zoomsizer, 1, wx.EXPAND, 0)
+        optionsizer.Add(thumbsizesizer, 1, wx.EXPAND, 0)
+        optionsizer.Add(self.fontbutton, 0, wx.ALL|wx.ADJUST_MINSIZE, 3)
+        optionsizer.Add(self.colourbutton, 0, wx.TOP|wx.LEFT|wx.ADJUST_MINSIZE, 3)
+        splitsizer.Add(optionsizer, 0, wx.EXPAND | wx.TOP|wx.LEFT, 5)
+        self.panel.SetAutoLayout(True)
```

Comment:
`SetAutoLayout` is redundant. `SetSizer` also calls it.

Reply:
Usually I put `SetSizer` or `SetSizerAndFit`. Fit can be a bit of a problem if you are being anal about appearance, so if you need a default, then try to use just `SetSizer` for 99% of cases. `SetSizer` should be fine for the demo. If not..., then explain why/etc...

Reply:
In most cases `SetSizerAndFit` has some unexpected side-effects so I usually suggest avoiding it unless you really know that you need it.

Reply:
Removed SetAutoLayout().  (Again, this is from the original ThumbnailCtrl code.)